### PR TITLE
[#157198021] Adjustes KCRConfigSvc to accomodate MIWA

### PR DESCRIPTION
**Pull request by pcsouza**
KCRConfigSvc renamed to KCEConfigSvc (KeyCardEncoder)
KCRConfig renamed to KabaConfig
added MiwaConfig

Added an enum to switch the active interface to the keycard system

### CODE REVIEW

File: csharp/HOLMS.Platform/HOLMS.Platform/Client/ApplicationClient.cs [hunk 1]
```diff
@@ -119,7 +119,7 @@ public class ApplicationClient : IDisposable, IApplicationClient {
         public ForeignPropertyConfigurationSvc.ForeignPropertyConfigurationSvcClient ForeignPropertyConfigurationSvc { get; protected set; }
         public GroupBookingMethodSvc.GroupBookingMethodSvcClient GroupBookingMethodSvc { get; protected set; }
         public HkPhoneCodeSvc.HkPhoneCodeSvcClient HkPhoneCodeSvc { get; protected set; }
-        public KCRConfigSvc.KCRConfigSvcClient KCRConfigSvc { get; protected set; }
+        public KCEConfigSvc.KCEConfigSvcClient KCRConfigSvc { get; protected set; }
```
[the remark this answers is not in the document]
Comment (eldavido) [reply csharp/HOLMS.Platform/HOLMS.Platform/Client/ApplicationClient.cs]: Btw, what is "KCE"? Keycard encoder?

File: proto/tenancy_config/active_kce_configuration.proto [hunk 2]
```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+
+package holms.types.tenancy_config;
+option csharp_namespace = "HOLMS.Types.TenancyConfig";
+option go_package = "tenancyconfig";
+
+enum ActiveKCEConfiguration {
```
Comment (eldavido) [inline proto/tenancy_config/active_kce_configuration.proto]: Can you be a little more explicit here and less "jargony"? Just call this "KeyCardBrand" or "ActiveKeyCardBrand".

File: proto/tenancy_config/kaba_config.proto [hunk 3]
```diff
@@ -4,7 +4,7 @@ package holms.types.tenancy_config;
 option csharp_namespace = "HOLMS.Types.TenancyConfig";
 option go_package = "tenancyconfig";
 
-message KCRConfig {
+message KabaConfig {
```
Comment (eldavido) [inline proto/tenancy_config/kaba_config.proto]: I'd prefer this be called `KabaEncoderConfig` or something indicating it's for the keycards. Kaba is just a brand which makes this pretty vague (just thinking about the next people who will have to make sense of this).

File: proto/tenancy_config/miwa_config.proto [hunk 4]
```diff
@@ -0,0 +1,12 @@
+syntax = "proto3";
+
+package holms.types.tenancy_config;
+option csharp_namespace = "HOLMS.Types.TenancyConfig";
+option go_package = "tenancyconfig";
+
+message MiwaConfig {
```
Comment (eldavido) [inline proto/tenancy_config/miwa_config.proto]: Same comment here, maybe `MiwaKeyCardConfig` or `MiwaCardEncoderConfig`?

File: proto/tenancy_config/rpc/kce_config_svc.proto [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+message KCESvcSetActiveKCEConfigRequest {
+	.holms.types.tenancy_config.ActiveKCEConfiguration active_kce_config = 1;
+}
+
+service KCEConfigSvc {
+	rpc Get(google.protobuf.Empty) 
+		returns (GetKCEConfigResponse);
+
+	rpc SetKabaConfig(.holms.types.tenancy_config.KabaConfig)
```
Comment (eldavido) [inline proto/tenancy_config/rpc/kce_config_svc.proto]: Can you comment a bit on the rationale for 3 rpc calls? It looks like the call pattern is "set active configuration", and then we store (potentially) both a MIWA and a KABA configuration separately?

What was the rationale for this design? Not knowing any better, I'd probably have designed it as a single configuration that could be switched between one or the other. Was the intention to have both in parallel to allow easy switching between the two? I think only one can be active at a time, right?

File: proto/tenancy_config/active_kce_configuration.proto (hunk 2 above)
Comment (pcsouza) [reply proto/tenancy_config/active_kce_configuration.proto]: Change to ActiveKeySystem

**Comment by pcsouza** [comment]
@eldavido Changes made PTAL

**Pull request merged**